### PR TITLE
infracost: improvements

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

@davegallant 

added -s and -w ldflags saving 2MB
added version ldflag
added pre and post check hooks
added completions
added installCheckPhase
added some more meta details

added myself as a maintainer if you don't mind

Once https://github.com/infracost/infracost/pull/862 is part of a release we can just set `export INFRACOST_SKIP_UPDATE_CHECK=true`

Completions embedded in the command (combined with the fact infracost is already added) closes #114402
There are going to be other urfav cli tools that could benefit from the completion but it should really be fixed in the cli tool itself

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/infracost/default.nix [hunk 1]
```diff
@@ -10,15 +10,56 @@ buildGoModule rec {
...
+    changelog = "https://github.com/infracost/infracost/releases/tag/v${version}";
     description = "Cloud cost estimates for Terraform in your CLI and pull requests";
-    homepage = "https://github.com/infracost/infracost";
+    longDescription = ''
+      Infracost shows hourly and monthly cost estimates for a Terraform project.
+      This helps developers, DevOps et al. quickly see the cost breakdown and
+      compare different deployment options upfront.
+    '';
     license = [ licenses.asl20 ];
     maintainers = [ maintainers.davegallant ];
```
Comment (davegallant) [inline pkgs/tools/misc/infracost/default.nix:64]: Are you going to add yourself as a maintainer?

Comment (06kellyjac) [reply pkgs/tools/misc/infracost/default.nix:64]: Ahh i thought i commit amend and forcepushed..
Ill fix this when i get back

Comment (06kellyjac) [reply pkgs/tools/misc/infracost/default.nix]: Fixed

File: pkgs/tools/misc/infracost/default.nix [hunk 2]
```diff
@@ -10,16 +10,57 @@ buildGoModule rec {
...
     description = "Cloud cost estimates for Terraform in your CLI and pull requests";
-    homepage = "https://github.com/infracost/infracost";
+    longDescription = ''
+      Infracost shows hourly and monthly cost estimates for a Terraform project.
+      This helps developers, DevOps et al. quickly see the cost breakdown and
+      compare different deployment options upfront.
+    '';
     license = [ licenses.asl20 ];
-    maintainers = [ maintainers.davegallant ];
+    maintainers = [ maintainers.davegallant jk ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/infracost/default.nix:64]:
```suggestion
    maintainers = with maintainers; [ davegallant jk ];
```

**Review by davegallant — APPROVED** [review]

**Pull request merged**